src/slur.cpp

```diff
@@ -168,126 +161,136 @@ int Slur::AdjustSlurCurve(Doc *doc, const ArrayOfCurveSpannedElements *spannedEl
     maxHeight = std::min(maxHeight, doc->GetDrawingOctaveSize(staffSize));
 
     maxHeight = currentHeight;
-    /*
-    bool hasReachedMaxHeight = false;
-
-    if (maxHeight > currentHeight) {
-        float maxRatio = 1.0;
-        float posXRatio = 1.0;
-        int posX;
-        for (itPoint = spannedElements->begin(); itPoint != spannedElements->end();) {
-
-            if (itPoint->second.second.x < p1->x) {
-                // itPoint = spannedElements->erase(itPoint);
-                ++itPoint;
-                continue;
-            }
-            if (itPoint->second.second.x > p2->x) {
-                // itPoint = spannedElements->erase(itPoint);
-                ++itPoint;
-                continue;
-            }
+    if (!spannedElements->empty()) return maxHeight;
 
-            y = BoundingBox::CalcBezierAtPosition(bezier, itPoint->second.second.x);
+    return 0;
+}
 
-            // Weight the desired height according to the x position if wanted
-            posXRatio = 1.0;
-            if (posRatio && (dist != 0)) {
-                posX = itPoint->second.second.x - p1->x;
-                if (posX > dist / 2) posX = p2->x - itPoint->second.second.x;
-                if (dist != 0) posXRatio = (float)posX / ((float)dist / 2.0);
+bool Slur::AdjustSlurPosition(
+    Doc *doc, FloatingCurvePositioner *curve, BezierCurve &bezierCurve, float &angle, bool forceBothSides)
+{
+    bool isNotAdjustable = false;
+    const int margin = doc->GetDrawingUnit(100);
+    auto [maxShiftLeft, maxShiftRight]
```

Comment:
Don't use `auto`

Reply:
I find structural bindings a little bit clearer, since we don't need to declare variables beforehand. But I'll change it to std::tie, since they cannot be used without 'auto'.